pkg/plugin/framework_methods.go

```diff
@@ -0,0 +1,610 @@
+			// 'otherPods' to mark it as already included.
+			delete(otherPods, uid)
+		}
+	}
+
+	// all that remains in otherPods is the pods that were not removed from iterating through
+	// tmpNode's pods.
+	var proposedNotInLocalState []podInfo
+	for _, p := range otherPods {
+		proposedNotInLocalState = append(proposedNotInLocalState, podInfo{
+			Namespace: p.Pod.Namespace,
+			Name:      p.Pod.Name,
+			UID:       p.Pod.UID,
+		})
+
+		pod, err := state.PodStateFromK8sObj(p.Pod)
+		if err != nil {
+			logger.Error(
+				"Ignoring extra Pod in Filter stage because extracting custom state failed",
+				reconcile.ObjectMetaLogField("Pod", p.Pod),
+				zap.Error(err),
+			)
+			continue
+		}
+
+		tmpNode.AddPod(pod)
+	}
+
+	// At this point:
+	// * oldNode has the actual local state for the node
+	// * tmpNode has the state as given by the filter pods
+	// * localNotInProposed are the pods in oldNode but not tmpNode
+	// * proposedNotInLocalState are the pods in tmpNode but not oldNode
+	//
+	// We'll use (another) Speculatively() to simultaneously show all these, plus the state
+	// resulting from adding the Pod to filter.
+	var canAddToNode bool
+	tmpNode.Speculatively(func(n *state.Node) (commit bool) {
+		n.AddPod(filterPod)
+		canAddToNode = !n.OverBudget()
+
+		var msg string
+		if canAddToNode {
+			msg = "Allowing Pod placement onto this Node"
+		} else {
+			msg = "Rejecting Pod placement onto this Node"
+		}
+		logger.Info(
+			msg,
+			zap.Object("Node", oldNode),
+			zap.Object("FilterNode", tmpNode),
+			zap.Object("FilterNodeWithPod", n),
+			zap.Object("Pod", filterPod),
+			zap.Any("LocalPodsNotInFilterState", localNotInProposed),
+			zap.Any("FilterPodsNotInLocalState", proposedNotInLocalState),
+		)
+
+		return false // don't commit. Doesn't really matter because we're operating on the temp node.
+	})
+	return canAddToNode
+}
+
+// Score allows our plugin to express which nodes should be preferred for scheduling new pods onto
+//
+// Even though this function is given (pod, node) pairs, our scoring is only really dependent on
+// values of the node. However, we have special handling for when the pod no longer fits in the node
+// (even though it might have during the Filter plugin) - we can't return a failure, because that
+// would cause *all* scheduling of the pod to fail, so we instead return the minimum score.
+//
+// The scores might not be consistent with each other, due to ongoing changes in the node. That's
+// ok, because nothing relies on strict correctness here, and they should be approximately correct
+// anyways.
+//
+// Score implements framework.ScorePlugin.
+func (e *AutoscaleEnforcer) Score(
+	ctx context.Context,
+	_state *framework.CycleState,
+	pod *corev1.Pod,
+	nodeName string,
+) (_ int64, status *framework.Status) {
+	ignored := e.state.config.ignoredNamespace(pod.Namespace)
+
+	e.metrics.IncMethodCall("NormalizeScore", pod, ignored)
```

Comment:
issue (blocking): `NormalizeScore` here looks like a copy-paste typo, I guess it should be `Score`?

suggestion: replace `NormalizeScore` with `Score` on lines 287, 289, 293

Reply:
Good catch, thanks! b47ba954b898ee519e7cc82b8d5f301b101b2ebe